doc/source/ray-core/objects.rst

```diff
@@ -128,41 +128,43 @@ If the current node's object store does not contain the object, the object is do
       assert(*results[1] == 1);
       assert(*results[2] == 2);
 
-Passing Objects by Reference
-----------------------------
+Passing Objects
+---------------
 
 Ray object references can be freely passed around a Ray application. This means that they can be passed as arguments to tasks, actor methods, and even stored in other objects. Objects are tracked via *distributed reference counting*, and their data is automatically freed once all references to the object are deleted.
 
-.. code-block:: python
+There are two different ways one can pass an object to a Ray task or method. Depending on the way an object is passed, Ray will decide whether to *de-reference* the object prior to task execution.
 
-    @ray.remote
-    def echo(x):
-        print(x)
+**Passing an object as a top-level argmuent**: When an object is passed directly as a top-level argument to a task, Ray will de-reference the object. This means that Ray will fetch the underlying data for all top-level object reference arguments, not executing the task until the object data becomes fully available.
```

Comment:
Nit: May want to add a warning that providing an object reference as a default argument is **not** de-referenced, even though it's technically a top-level argument.

Reply:
Hm what do you mean by that? Is it closure capture style? Like `def f(x=x_ref)`?

I can add a section for closure capture of refs.

Reply:
Yep closure capture style!